### DISCUSSION TITLE
what happened to ClientTimestamp?

### DISCUSSION THREAD

**Discussion by springy76** [datalust/seq-tickets#2115] [Help]
Starting 3 weeks ago (most likely with first 2024.1 version) not a single seq log entry has the ClientTimestamp property anymore. Is there a setting for this which needs to get enabled again?

**Comment by nblumhardt** [comment]
Hi @springy76; Seq 2024.1 included some rationalization of different ingestion paths and no longer provides `ClientTimestamp`. It'd be good to know where/how this fits into your scenario. Thanks!

**Reply by springy76** [reply]
As messages always get delivered in batches (using Serilog) the ClientTimestamp helped understanding the real order of messages when there is REST communication involved between multiple services, all logging to same SEQ instance.

Might be much simpler with the new tracing capabilities, but this has to be implemented and tested first.

**Reply by nblumhardt** [reply]
Thanks for your reply. Do you also have significant clock differences between machines to deal with (i.e. are you using server timestamps because of incorrect client clocks)?

Tracing will show the correct hierarchy of calls between services, so that could definitely help, but the timeline view that's fairly central to the tracing experience will still rely on accurate client clocks in order to display relative span start and end times correctly.

If you're interested in digging into tracing further, and need any help getting it integrated, please feel free to send mail to support@datalust.co, happy to field questions there.
